From d12250e94b840d90625347e59c16f4bf7292050a Mon Sep 17 00:00:00 2001
From: Thomas White <tomwhite@apache.org>
Date: Wed, 23 Jan 2013 11:06:39 +0000
Subject: [PATCH 1265/1357] MR1: MAPREDUCE-4929. mapreduce.task.timeout is ignored. Contributed by Sandy Ryza.

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/branches/branch-1@1437343 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit f92d8753bc4ab98f68f6556fa802bcd11e1cabfd)

Reason: Support/customer request
Author: Sandy Ryza
Ref: CDH-9078
---
 .../org/apache/hadoop/mapred/TaskTracker.java      |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/TaskTracker.java b/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/TaskTracker.java
index 8a12b7e..02264b1 100644
--- a/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/TaskTracker.java
+++ b/hadoop-mapreduce1-project/src/mapred/org/apache/hadoop/mapred/TaskTracker.java
@@ -2942,7 +2942,11 @@ public class TaskTracker implements MRConstants, TaskUmbilicalProtocol,
       this.localJobConf = lconf;
       keepFailedTaskFiles = localJobConf.getKeepFailedTaskFiles();
       taskTimeout = localJobConf.getLong("mapred.task.timeout", 
-                                         10 * 60 * 1000);
+                                         Integer.MIN_VALUE);
+      if (taskTimeout == Integer.MIN_VALUE) {
+        taskTimeout = localJobConf.getLong("mapreduce.task.timeout", 
+            10 * 60 * 1000);
+      }
       if (task.isMapTask()) {
         debugCommand = localJobConf.getMapDebugScript();
       } else {
-- 
1.7.0.4

